projects/archiver/zipper.ts

```diff
@@ -0,0 +1,66 @@
+import { S3, SharedIniFileCredentials } from 'aws-sdk'
+import { PassThrough } from 'stream'
+import archiver = require('archiver')
+
+const creds = process.env.arn
+    ? {}
+    : {
+          credentials: new SharedIniFileCredentials({ profile: 'frontend' }),
+      }
+const s3 = new S3({
+    region: 'eu-west-1',
+    ...creds,
+})
+
+const notNull = <T>(value: T | null | undefined): value is T =>
+    value !== null && value !== undefined
+
+const removeInitial = (s: string, remove: string) => {
+    if (!s.startsWith(remove)) return s
+    return s.substr(remove.length)
+}
+
+export const zip = async (name: string, root: string, prefix: string) => {
+    const output = new PassThrough()
+    const upload = s3
+        .upload({
+            Bucket: 'editions-store',
+            Key: `zips/${name}.zip`,
+            Body: output,
+            ACL: 'public-read',
+        })
+        .promise()
+
+    const objects = await s3
+        .listObjectsV2({
+            Bucket: 'editions-store',
+            Prefix: `${root}/${prefix}`,
+        })
+        .promise()
+    const files = (objects.Contents || []).map(obj => obj.Key).filter(notNull)
+
+    console.log('Got file names')
+
+    const archive = archiver('zip')
+    archive.on('warning', err => {
+        console.error('Error in attempting to compress', err)
+    })
+    archive.pipe(output)
+    await Promise.all(
+        files.map(async file => {
+            console.log(`getting ${file}`)
+            const s3response = await s3
+                .getObject({ Bucket: 'editions-store', Key: file })
```

Comment:
I wonder whether `editions-store` could at least be a constant as it's used a few places, if not wrap a tiny `EditionsStore` singleton with `get` and `put` methods?

Reply:
Should be taking it from the cdk params now anyway. Will fix.